python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -193,6 +194,32 @@ def get_in_progress_asset_subset(self, *, asset_key: AssetKey) -> ValidAssetSubs
 
         return ValidAssetSubset(asset_key=asset_key, value=value)
 
+    @cached_method
+    def get_failed_asset_subset(self, *, asset_key: AssetKey) -> ValidAssetSubset:
+        """Returns an AssetSubset representing the subset of the asset that failed to be
+        materialized its most recent run.
+        """
+        partitions_def = self.asset_graph.get(asset_key).partitions_def
+        if partitions_def:
+            cache_value = self._get_updated_cache_value(asset_key=asset_key)
+            if cache_value is None:
+                value = partitions_def.empty_subset()
+            else:
+                value = cache_value.deserialize_failed_partition_subsets(partitions_def)
+        else:
+            # ideally, unpartitioned assets would also be handled by the asset status cache
```

Comment:
Can we start marking cases like these with a greppable string (linear task identifier of some sort)? So when we figure out a nicer API we can easily find callsites.